### DISCUSSION TITLE
Property [$email] not found on component: [user]

### DISCUSSION THREAD

**Discussion by tariq-khan9** [livewire/livewire#5357] [1. Help]
i have used real-time validation of livewire but when i type and the real-time validation triggers, it gives the following error. tried everyting but dont know what is the real cause. 
Livewire \ Exceptions \ PropertyNotFoundException
PHP 8.1.2-1ubuntu2.8
9.39.0
Property [$email] not found on component: [user]

this is my blade file structure: user.blade-> <livewire:user>-> @include('livewire.adduser')

![Screenshot from 2022-11-13 12-10-11](https://user-images.githubusercontent.com/110604408/201510445-0f3b2c25-e5ef-4b6e-b367-8ca68dd9060d.png)
![Screenshot from 2022-11-13 12-11-57](https://user-images.githubusercontent.com/110604408/201510457-c9a29a2f-33d6-4362-9e0e-877de9a01b91.png)

**Comment by alfakiat** [comment]
Try to move all attributes:
`$name` `$email` `$password`

 and all functions: 
`rules()` 
`updated()` 

to `[User]` livewire component.

because the actions occur on livewire components not on UserController.

**Reply by joshhanley** [reply]
@alfakiat is correct, this code should be in a Livewire component, not in your UserController. Fix that up and it should work 🙂
